frontend/src/components/pages/RegistrantExperience/PersonalInfo/index.tsx

```diff
@@ -1,26 +1,110 @@
 import React from "react";
-import { Box, Checkbox, Text } from "@chakra-ui/react";
+import { SmallAddIcon } from "@chakra-ui/icons";
+import { Box, Button, Divider, Text, useToast, VStack } from "@chakra-ui/react";
+import { RegistrantExperienceCamper } from "../../../../types/CamperTypes";
+import { usePersonalInfoHook } from "./personalInfoReducer";
+import CamperCard from "./CamperCard";
+import { PersonalInfoActions } from "../../../../types/PersonalInfoTypes";
+import ContactCard from "./ContactCard";
+import { CampSession } from "../../../../types/CampsTypes";
 
 type PersonalInfoProps = {
-  isChecked: boolean;
-  toggleChecked: () => void;
+  nextBtnRef: React.RefObject<HTMLButtonElement>;
+  campers: RegistrantExperienceCamper[];
+  campName: string;
+  campSessions: CampSession[];
+  setCampers: React.Dispatch<
+    React.SetStateAction<RegistrantExperienceCamper[]>
+  >;
+};
+
+const checkSpaceAvailable = (
+  campSessions: CampSession[],
+  campers: RegistrantExperienceCamper[],
+): boolean => {
+  // validate whether there is enough space for a new camper
+  let spaceAvailable = 100000000;
+  /* eslint-disable-next-line */
+  for (const campSession of campSessions) spaceAvailable = Math.min(spaceAvailable,campSession.capacity - campSession.campers.length - campers.length);
+  return spaceAvailable > 0;
 };
 
 const PersonalInfo = ({
-  isChecked,
-  toggleChecked,
+  nextBtnRef,
+  campers,
+  campName,
+  campSessions,
+  setCampers,
 }: PersonalInfoProps): React.ReactElement => {
+  const toast = useToast();
+  const setPersonalInfo = usePersonalInfoHook(setCampers);
```

Comment:
If I understand it correctly, we use `setPersonalInfo` to dispatch actions. If so, could we rename this to `dispatchPersonalInfoAction` or something a bit more verbose?

Reply:
Good point. Let me change it.